lib/clone-pattern.js

```diff
@@ -0,0 +1,97 @@
+var path = require('path');
+var fs = require('fs-extra');
+var utility = require('./utility');
+
+/*
+ * Clone a pattern's entire directory to destination directory
+ *
+ * @param {Array} paths  path info on original pattern
+ * @param {Object} options  pattern-importer options
+ * @param {String} destDir  the top of destination directory for the cloned pattern
+*/
+exports.clonePattern = function (paths, options, destDir, callback) {
+  'use strict';
+
+  // make sure the destination directory is a string and exists
+  destDir = destDir || './patterns';
+  if (typeof destDir !== 'string') {
+    throw new Error('Destination directory must be a string.');
+  }
+
+  // open the pattern's data file
+  var patternYml = fs.readFileSync(path.join(paths.folder, options.dataFileName), {encoding: 'utf8'});
+
+  // get metadata from pattern.yml file
+  var patternObject = utility.convertYamlToObject(patternYml);
+
+  // get the pattern's category directory
+  var patternCategoryPath = utility.getCategoryPath(patternObject, options);
+
+  // determine the cloned pattern's destination directory inside the local patterns directory
+  var clonedPatternDest = path.resolve(path.join(destDir, patternCategoryPath, paths.directory));
+
+  // check if a directory already exists in our destination
+  fs.open(clonedPatternDest,'r',function(err,fd){
+    if (err && err.code == 'ENOENT') {
+
+      // copy the entire pattern directory, as-is, to our destination directory
+      fs.copy(paths.folder, clonedPatternDest, function (err) {
+        if (err) return console.error(err);
+
+        // success in cloning the directory
+        console.log(paths.directory + ' pattern cloned into ' + clonedPatternDest);
```

Comment:
Hey @e2tha-e 

cc: @conortm 

That is the point of [this logging story](https://rally1.rallydev.com/#/26707410139ud/detail/userstory/43034541458) and task which are assigned to you for this iteration (info email sent last thursday).


Once we have proper logging in place, it will be replacing the console.logs/throws/etc which exist throughout these scripts.

Until then, please ignore console.logs in any PRs.